### PR TITLE
RELATED: RAIL-3849 - Fix parent-child filtering issues

### PR DESCRIPTION
**Pull request by plavemar**
- added filter background while filtering by parent
- update all descendants of the parent attribute filter

JIRA: RAIL-3849

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: libs/sdk-ui-dashboard/src/model/store/filterContext/filterContextSelectors.ts [hunk 1]
```diff
@@ -159,3 +160,29 @@ export const selectFilterContextAttributeFilterIndexByLocalId = createMemoizedSe
         attributeFilters.findIndex((filter) => filter.attributeFilter.localIdentifier === localId),
     ),
 );
+
+/**
+ * Creates a selector for selecting all descendants of the attribute filter with given localId.
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/model/store/filterContext/filterContextSelectors.ts:165]: Please add a release tagy, probably `@internal`

File: libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx [hunk 2]
```diff
@@ -489,7 +495,7 @@ export const AttributeFilterButtonCore: React.FC<IAttributeFilterButtonProps> =
                 setPlaceholderValue(emptyFilter);
             }
             const isInverted = isNegativeAttributeFilter(currentFilter);
-            props.onApply?.(emptyFilter, isInverted);
+            // props.onApply?.(emptyFilter, isInverted);
```
Comment (no23reason) [inline libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx:498]: Forgotten commented out code?

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1963

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1707/ : SUCCESS in 58s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1988/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/785/ : SUCCESS in 14m 38s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7700/ : FAILURE in 10m 22s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1963

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1708/ : SUCCESS in 1m 15s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1989/ : SUCCESS in 22s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/786/ : SUCCESS in 15m 57s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7704/ : SUCCESS in 17m 00s

**Pull request merged**